libraries/abstractions/pkcs11/mbedtls/iot_pkcs11_mbedtls.c

```diff
@@ -1025,58 +1026,57 @@ static CK_RV prvDeleteObjectFromList( CK_OBJECT_HANDLE xAppHandle )
  */
 static CK_RV prvAddObjectToList( CK_OBJECT_HANDLE xPalHandle,
                                  CK_OBJECT_HANDLE_PTR pxAppHandle,
-                                 const uint8_t * pcLabel,
-                                 size_t xLabelLength )
+                                 const CK_BYTE_PTR pcLabel,
+                                 CK_ULONG xLabelLength )
 {
-    CK_RV xResult = CKR_OK;
-    BaseType_t xGotSemaphore;
+    CK_RV xResult = CKR_HOST_MEMORY;
 
     /* See explanation in prvCheckValidSessionAndModule for this exception. */
     /* coverity[misra_c_2012_rule_10_5_violation] */
     CK_BBOOL xObjectFound = ( CK_BBOOL ) CK_FALSE;
-    int32_t lInsertIndex = -1;
-    int32_t lSearchIndex = pkcs11configMAX_NUM_OBJECTS - 1;
-
-    xGotSemaphore = xSemaphoreTake( xP11Context.xObjectList.xMutex, portMAX_DELAY );
+    uint32_t ulSearchIndex = 0;
 
-    if( xGotSemaphore == pdTRUE )
+    if( pdTRUE == xSemaphoreTake( xP11Context.xObjectList.xMutex, portMAX_DELAY ) )
```

Comment:
What is the reason for this change? I think it is easier to debug if the return value is stored in a local variable.

Reply:
I like the conciseness, and not having to declare a variable. I don't mind reverting this